packages/api/src/providers/tenovi.ts

```diff
@@ -27,6 +31,106 @@ export class Tenovi extends Provider {
     });
   }
 
+  /**
+   * Disconnects all connected devices from the user's Tenovi Gateway.
+   * Removes Tenovi from the user's ProviderMap.
+   *
+   * @param connectedUser The user to disconnect the device from
+   */
+  async revokeProviderAccess(connectedUser: ConnectedUser): Promise<void> {
+    if (connectedUser.providerMap?.tenovi?.connectedDeviceIds) {
+      const res = await Promise.allSettled(
+        connectedUser.providerMap?.tenovi?.connectedDeviceIds?.map(async deviceId => {
+          await this.disconnectDevice(connectedUser, deviceId, false);
+        })
+      );
+
+      const rejected = res.filter(r => r.status === "rejected");
+      if (rejected.length) {
+        throw new Error(
+          `Failed to disconnect ${rejected.length} devices from Tenovi Gateway. User: ${connectedUser.id}`
+        );
```

Comment:
Let's avoid adding dynamic information to the error description - if this makes its way to Sentry, it won't be able to group them together.

We could use `MetriportError` and store details on the third parameter to its constructor.